tests/Middlewares/MiddlewareTest.php

```diff
@@ -369,6 +370,71 @@ public function testFindologicIsTriggeredIfProperlyConfigured(
         $middleware->before($requestMock);
     }
 
+    public function languageProvider()
+    {
+        return [
+            'Default language used' => [
+                'defaultLanguage' => 'de',
+                'usedLanguage' => 'de',
+                'expectedLanguagePath' => ''
+            ],
+            'Non-default language used' => [
+                'defaultLanguage' => 'de',
+                'usedLanguage' => 'fr',
+                'expectedLanguagePath' => '/fr'
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider languageProvider
+     */
+    public function testLanguagePath($defaultLanguage, $usedLanguage, $expectedLanguagePath)
+    {
+        $webstoreConfigMock = $this->getMockBuilder(WebstoreConfiguration::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $webstoreConfigMock->defaultLanguage = $defaultLanguage;
+
+        $webstoreConfigurationRepositoryMock =
+            $this->getMockBuilder(WebstoreConfigurationRepositoryContract::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $webstoreConfigurationRepositoryMock->expects($this->once())
+            ->method('getWebstoreConfiguration')
+            ->willReturn($webstoreConfigMock);
+
+        $localizationRepositoryMock = $this->getMockBuilder(LocalizationRepositoryContract::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $localizationRepositoryMock->expects($this->once())
+            ->method('getLanguage')
+            ->willReturn($usedLanguage);
+
+        global $classInstances;
```

Comment:
Can we reset the state of this in the `tearDown` to just be an empty array after the test has run?